### PR TITLE
improve `VALIDATION_WEBHOOK_CONFIG_NAME` environment variable description

### LINKED ISSUE #42681: VALIDATION_WEBHOOK_CONFIG_NAME env variable getting ignored
### Bug Description

We have observed that VALIDATION_WEBHOOK_CONFIG_NAME env which is documented is not actually getting ignored in the code.

https://github.com/istio/istio/blob/3fcff36ddc5e69ed20755c6385d44eb1a0e50505/pilot/pkg/features/pilot.go#L451-L452
While the code here checks for the env variable here
https://github.com/istio/istio/blob/89c548304627c7a2bed1216ce2b180560a8d5a67/pilot/pkg/bootstrap/validation.go#L42-L49

The actual code which patches validationWebhook uses label selector instead and the VALIDATION_WEBHOOK_CONFIG_NAME is ignored

https://github.com/istio/istio/blob/89c548304627c7a2bed1216ce2b180560a8d5a67/pkg/webhooks/validation/controller/controller.go#L203

### Version

```prose
1.15
```

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by wulianglongrd**
**Please provide a description of this PR:**

improve `VALIDATION_WEBHOOK_CONFIG_NAME` environment variable description. see also #42681

### CODE REVIEW

File: pilot/pkg/features/pilot.go [hunk 1]
```diff
@@ -448,8 +448,9 @@ var (
 	InjectionWebhookConfigName = env.Register("INJECTION_WEBHOOK_CONFIG_NAME", "istio-sidecar-injector",
 		"Name of the mutatingwebhookconfiguration to patch, if istioctl is not used.").Get()
 
-	ValidationWebhookConfigName = env.Register("VALIDATION_WEBHOOK_CONFIG_NAME", "istio-istio-system",
-		"Name of the validatingwebhookconfiguration to patch. Empty will skip using cluster admin to patch.").Get()
+	EnableValidatingWebhookConfigPatch = env.Register("PILOT_ENABLE_VALIDATING_WEBHOOK_CONFIG_PATCH", true,
```
Comment (howardjohn) [inline pilot/pkg/features/pilot.go:451]: This is a breaking change, can we keep the old one? and just update the docs.

Comment (wulianglongrd) [reply pilot/pkg/features/pilot.go]: i think maybe, all changes have been reverted, and only upate the docs.

File: pilot/pkg/features/pilot.go [hunk 2]
```diff
@@ -448,8 +448,10 @@ var (
 	InjectionWebhookConfigName = env.Register("INJECTION_WEBHOOK_CONFIG_NAME", "istio-sidecar-injector",
 		"Name of the mutatingwebhookconfiguration to patch, if istioctl is not used.").Get()
 
+	// ValidationWebhookConfigName see #42681
 	ValidationWebhookConfigName = env.Register("VALIDATION_WEBHOOK_CONFIG_NAME", "istio-istio-system",
-		"Name of the validatingwebhookconfiguration to patch. Empty will skip using cluster admin to patch.").Get()
+		"If not empty, the controller will automatically patch validatingwebhookconfiguration when the CA certificate changes. "+
+			"Only works when has k8s configStore or has kubeRegistry.").Get()
```
Comment (hzxuzhonghu) [inline pilot/pkg/features/pilot.go:454]:
```suggestion
			"Only works in kubernetes environment.").Get()
```

**Comment by zhlsunshine** [comment]
What's the really change for this PR? It seems that the changes does not match the description in issue #42681.

**Comment by wulianglongrd** [comment]
> What's the really change for this PR? It seems that the changes does not match the description in issue #42681.

we think that using the label patch is as expected, but the description of the `VALIDATION_WEBHOOK_CONFIG_NAME` environment variable is incorrect.

**Comment by zhlsunshine** [comment]
> we think that using the label patch is as expected, but the description of the `VALIDATION_WEBHOOK_CONFIG_NAME` environment variable is incorrect.

Yeah, I noticed that you want to change the env variable from string to bool, right? But I also see that you revert it. So just want to know the exact changes.

File: pilot/pkg/features/pilot.go [hunk 3]
```diff
@@ -448,8 +448,10 @@ var (
 	InjectionWebhookConfigName = env.Register("INJECTION_WEBHOOK_CONFIG_NAME", "istio-sidecar-injector",
 		"Name of the mutatingwebhookconfiguration to patch, if istioctl is not used.").Get()
 
+	// ValidationWebhookConfigName see #42681
```
Comment (howardjohn) [inline pilot/pkg/features/pilot.go:451]: No need to link to an issue, just document the field usage here?

**Review by howardjohn — APPROVED** [review]

**Comment by wulianglongrd** [comment]
need to add a `release-notes-none` label

**Review by hzxuzhonghu — APPROVED** [review]
/release-note-none

**Pull request merged**